controllers/front/product.php

```diff
@@ -28,36 +28,42 @@ class ShoppingfeedProductModuleFrontController extends \ModuleFrontController
 {
     protected $sfToken = null;
 
-    public function initContent()
+    public function init()
     {
-        $token = Tools::getValue('token');
-        if (empty($token)) {
-            $token = (new ShoppingfeedToken())->getDefaultToken();
-        } else {
-            $token = (new ShoppingfeedToken())->findByToken($token);
-        }
+        parent::init();
+
+        $this->sfToken = (new ShoppingfeedToken())->findByToken(Tools::getValue('token', ''));
+    }
 
-        if ($token === false) {
-            exit();
+    public function viewAccess()
+    {
+        if (empty($this->sfToken)) {
```

Comment:
perhaps better in checkAccess() method ?

Reply:
Yes, checkAccess() is better. I have changed it